pkg/apis/core/helper/shoot.go

```diff
@@ -125,6 +125,18 @@ func GetShootETCDEncryptionKeyRotationPhase(credentials *core.ShootCredentials)
 	return ""
 }
 
+// IsShootETCDEncryptionKeyRotationSingleOperation returns whether the current ETCD encryption key rotation is a single
+// operation rotation or not.
+func IsShootETCDEncryptionKeyRotationSingleOperation(credentials *core.ShootCredentials) bool {
+	if credentials != nil &&
+		credentials.Rotation != nil &&
+		credentials.Rotation.ETCDEncryptionKey != nil &&
+		credentials.Rotation.ETCDEncryptionKey.IsSingleOperationRotation != nil {
+		return *credentials.Rotation.ETCDEncryptionKey.IsSingleOperationRotation
```

Comment:
Wdyt about naming the new field something like this to indicate exactly what is happening: automtically triggering a complete operation after the preparing operation has finished?
```suggestion
		return *credentials.Rotation.ETCDEncryptionKey.AutoCompleteAfterPrepared
```